lib/mp4/transmuxer.js

```diff
@@ -546,104 +547,170 @@ Transmuxer = function(options) {
     videoTrack,
     audioTrack,
     packetStream, parseStream, elementaryStream,
-    aacStream, h264Stream,
+    adtsStream, h264Stream,aacStream,
     videoSegmentStream, audioSegmentStream, captionStream,
-    coalesceStream;
+    coalesceStream,
+    aacfile;
+
+  this.setupAACPipeline = function() {
+    this.metadataStream = new AacStream.MetadataStream();
+    options.metadataStream = this.metadataStream;
+
+    // set up the parsing pipeline
+    aacStream = new AacStream.AacStream();
+    adtsStream = new AdtsStream();
+    coalesceStream = new CoalesceStream(options);
+
+    aacStream.pipe(adtsStream);
+    aacStream.pipe(this.metadataStream);
+    this.metadataStream.pipe(coalesceStream);
+
+    this.metadataStream.on('timestamp', function(frame) {
+      aacStream.setTimestamp(frame.timestamp);
+    });
+    this.addAACListener();
+  };
 
-  Transmuxer.prototype.init.call(this);
-  options = options || {};
 
-  this.baseMediaDecodeTime = options.baseMediaDecodeTime || 0;
+  this.addAACListener = function() {
```

Comment:
Our caps standard for this project means that this should be called `addAacListener `.